dace/transformation/dataflow/vectorization.py

```diff
@@ -1,13 +1,138 @@
 # Copyright 2019-2021 ETH Zurich and the DaCe authors. All rights reserved.
 """ Contains classes that implement the vectorization transformation. """
-from dace import data, dtypes, registry, symbolic, subsets
-from dace.sdfg import nodes, SDFG, SDFGState, propagation
+from re import M
+import re
+from numpy import core
+import sympy
+
+from numpy.lib.arraysetops import isin
+from dace import data, dtypes, registry, symbolic, subsets, symbol
+from dace.frontend.octave.lexer import raise_exception
+from dace.sdfg import nodes, SDFG, SDFGState, propagation, InterstateEdge
 from dace.sdfg import utils as sdutil
 from dace.sdfg.scope import ScopeSubgraphView
 from dace.transformation import transformation
 from dace.transformation.helpers import replicate_scope
-from dace.properties import Property, make_properties
+from dace.properties import Property, make_properties, EnumProperty, SymbolicProperty
 import itertools
+import dace.transformation.dataflow.vectorization_infer_types as infer_types
+import dace.sdfg.analysis.vector_inference as vector_inference
+import dace.codegen.targets.sve.util as sve_util
+import dace.frontend.operations
+
+
+# Find the innermost state in which the node is
+# There is probably a better solution than this
+def get_innermost_state_for_node(sdfg: SDFG, node):
```

Comment:
Why "innermost"? Can't the node only be in one state?

Reply:
When nested SDFGs are involved, I observed that it somestimes occurs that some variables (e.g. arrays) can only be accessed via the inner State/SDFG. This is a somewhat hacky way to find the right SDFG/state.